contracts/multisig-improved/src/action_types/external_module.rs

```diff
@@ -0,0 +1,80 @@
+use multiversx_sc_modules::transfer_role_proxy::PaymentsVec;
+
+multiversx_sc::imports!();
+
+pub type ModuleId = AddressId;
+
+mod external_module_proxy {
+    use multiversx_sc_modules::transfer_role_proxy::PaymentsVec;
+
+    multiversx_sc::imports!();
+
+    #[multiversx_sc::proxy]
+    pub trait ExternalModuleProxy {
+        #[view(canExecute)]
+        fn can_execute(
+            &self,
+            sc_address: ManagedAddress,
+            original_caller: ManagedAddress,
```

Comment:
Instead of the original caller we should send the address from the signature to the modules

Reply:
Fixed.